### PR TITLE
fix: SMB2.1 WPTS conformance improvements (150/335 passing)

### PR DESCRIPTION
**Pull request by marmos91**
## Summary

Comprehensive SMB2.1 conformance improvements verified against Microsoft Windows Protocol Test Suite (WPTS). This branch improves WPTS BVT pass rate from baseline to **150 passing tests** with 90 known failures (all unimplemented features) and 0 new failures.

### Key fixes:

**Protocol correctness:**
- Fix QUERY_INFO response OutputBufferOffset (9 → 8, per MS-SMB2 StructureSize convention) — resolved 38+ test failures
- Fix SET_INFO FileBasicInformation buffer validation (40 bytes per MS-FSCC 2.4.7)
- Add STATUS_INFO_LENGTH_MISMATCH for undersized QUERY_INFO output buffers
- Fix SMB1 negotiate dialect parsing and wildcard response (0x02FF)
- Fix error response body encoding for non-success status codes

**FileInfo and directory operations:**
- Implement FileModeInformation, FilePositionInformation, FileAlignmentInformation
- Implement FileIdInformation for NTFS compatibility
- Fix FileNormalizedNameInformation path format (no leading backslash)
- Fix DOS_STAR (<) wildcard matching to consume through last dot
- Implement proper SingleEntry flag continuation in QueryDirectory
- Add SET_INFO attribute validation (DIRECTORY on file, TEMPORARY on dir)
- Strip NTFS stream suffixes (::$DATA, ::$INDEX_ALLOCATION) in CREATE
- Generate 8.3 short names for FileBothDirInfo/FileIdBothDirInfo entries

**Timestamp handling:**
- Handle FILETIME sentinels (0, -1, -2) in SET_INFO per MS-FSA 2.1.5.14.2
- Implement timestamp freeze/unfreeze semantics
- Support ChangeTime and CreationTime in SET_INFO operations
- Report NTFS-compatible filesystem sector/cluster sizes

**Lock conformance:**
- Implement SMB2 byte-range lock semantics
- Shared locks correctly block writes from ALL sessions (including lock holder)

### Remaining 90 known failures (all unimplemented features):
- VHD/RSVD: 19, ChangeNotify: 17, NTFS-FsCtl: 11, ADS: 9
- SWN: 5, DurableHandle/OpLock/Leasing: 4, Timestamp auto-update: 4
- SQoS: 3, FsInfo flags: 3, Timestamp sentinel edge cases: 3
- DFS: 2, NamedPipe: 2

## Test plan

- [x] All Go unit tests pass (`go test ./...`)
- [x] WPTS BVT conformance: **150 PASS**, 90 KNOWN, 0 NEW failures
- [x] `go vet ./...` clean
- [x] `go fmt ./...` applied
- [x] No regressions from previous passing tests

### CODE REVIEW

File: test/smb-conformance/Dockerfile.dittofs [hunk 1]
```diff
@@ -50,9 +50,8 @@ COPY --from=builder --chown=65532:65532 /build/dfsctl /app/dfsctl
 RUN mkdir -p /data/metadata /data/content /data/cache /config && \
     chown -R 65532:65532 /data /config
 
-USER 65532:65532
-
-EXPOSE 12445/tcp 8080/tcp
+# Run as root to allow binding to privileged port 445 (required by WPTS)
+EXPOSE 445/tcp 8080/tcp
```
[the remark this answers is not in the document]
Comment (marmos91) [reply test/smb-conformance/Dockerfile.dittofs]: Fixed: switched from running as root to using `setcap cap_net_bind_service=+ep` on the dfs binary, with `USER 65532:65532` to run as the non-root dittofs user.

File: internal/adapter/smb/response.go [hunk 2]
```diff
@@ -61,6 +61,12 @@ func ProcessSingleRequest(
 		return SendErrorResponse(reqHeader, types.StatusInternalError, connInfo)
 	}
 
+	// Per [MS-SMB2] 3.3.5.16: CANCEL must not send a response.
+	// Handlers return nil result to indicate "no response should be sent".
+	if result == nil {
```
[the remark this answers is not in the document]
Comment (marmos91) [reply internal/adapter/smb/response.go]: Fixed: added warning log when a non-CANCEL handler returns nil result.

File: internal/adapter/smb/v2/handlers/stub_handlers.go [hunk 3]
```diff
@@ -393,6 +407,204 @@ func (h *Handler) OplockBreak(ctx *SMBHandlerContext, body []byte) (*HandlerResu
...
+		binary.LittleEndian.PutUint64(output[24:32], 0)                                      // Usn
+		binary.LittleEndian.PutUint64(output[32:40], 0)                                      // TimeStamp
+		binary.LittleEndian.PutUint32(output[40:44], 0)                                      // Reason
+		binary.LittleEndian.PutUint32(output[44:48], 0)                                      // SourceInfo
+		binary.LittleEndian.PutUint32(output[48:52], 0)                                      // SecurityId
+		binary.LittleEndian.PutUint32(output[52:56], fileAttrs)                              // FileAttributes
+		binary.LittleEndian.PutUint16(output[56:58], uint16(len(fileNameBytes)))             // FileNameLength
+		binary.LittleEndian.PutUint16(output[58:60], v2FixedSize)                            // FileNameOffset
+		copy(output[v2FixedSize:], fileNameBytes)
+	}
```
[the remark this answers is not in the document]
Comment (marmos91) [reply internal/adapter/smb/v2/handlers/stub_handlers.go]: Added comment documenting that Usn/TimeStamp/Reason/SourceInfo/SecurityId are stub zeros, sufficient for WPTS but would need real values for USN journal functionality.

File: internal/adapter/smb/response.go [hunk 4]
```diff
@@ -272,10 +293,41 @@ func SendAsyncChangeNotifyResponse(sessionID, messageID uint64, response *handle
...
+			}
+			if end < len(dialects) {
+				name := string(dialects[:end])
+				if name == "SMB 2.???" {
+					responseDialect = types.SMB2DialectWild
+				}
+				dialects = dialects[end+1:] // skip past null terminator
+			} else {
+				break
+			}
```
[the remark this answers is not in the document]
Comment (marmos91) [reply internal/adapter/smb/response.go]: Fixed: added bounds check — if no null terminator is found within the buffer, we now break out of the parsing loop instead of continuing.
